File: Rust/src/longest_common_prefix.rs
pub fn longest_common_prefix(strs: Vec<String>) -> String {
    let mut prefix = strs.first().unwrap().to_string();
    for v in strs.iter().skip(1) {
        while v.find(prefix.as_str()) != Some(0) {
            prefix.pop();
            if prefix.is_empty() {
                return "".to_string();
            }
        }
    }
    prefix.to_string()
}

File: Ruby/lib/longest_common_prefix.rb
# frozen_string_literal: true

# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  prefix = strs[0]
  for i in 1...strs.length
    while strs[i].index(prefix) != 0
      prefix = prefix[0, prefix.length - 1]
      return '' if prefix.empty?
    end
  end
  prefix
end
